[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/googlecolab/colabtools/blob/master/notebooks/colab-github-demo.ipynb)

# Description

Code below shows how accurate five midpoint formula estimates derivative. We can specify the error bound for the estimate by comparing with exact estimate.

I choose sin^2x 5th derivative for calculation.

# Formula variable

x is the 0.1 radian
For h, choose h=0.01

Fifth derivative of sin^2(x) is 32sin(x)cos(x)
for the error I choose value between 0.08 and 0.12 and i choose 0.08

In [2]:
import numpy as np

#value for h on the formula 0.01
h=0.01

#radian value x=0.1
x=0.1

#determining the error part of the formula
error=((32*np.sin(0.08)*np.cos(0.08))*((0.01)**5))/5

#function gives first derivative using five point midpoint formula
f=((1/(12*h))*(((np.sin(x-(2*h)))**2)-(8*(np.sin((x-h)))**2)+(8*(np.sin(x+h))**2)-(np.sin(x+(2*h)))**2))+error
print("Five midpoint formula gives us answer ", f)

#exact value using numpy 
exactValue = (np.cos(.1))*(np.sin(.1))*2
print ("Exact value", exactValue)

#the remaining error
print("difference is", exactValue-f)


Five midpoint formula gives us answer  0.19866932978652377
Exact value 0.19866933079506124
difference is 1.0085374757551335e-09


# Analysis
Using the five point mid-point formula i got 0.19866932978652377 

Using numpy the first derivative 2sin(x)cos(x)= 0.19866933079506124

The error we get is=1.0085374757551335e-09

I have tried using 0.09,0.1, 0.11, 0.12 for the error. But it gave me the biggest error. So I used 0.8 for the error as it gave me the smallest difference. The fivepoint midpoint formula gave me fairly close answer to the exact answer. The formula is somewhat accurate for calculation, but not reliable for exact calculation even with adding the extra error formula.  

![Title](meme.jpg)